bank-sdk/example-app/src/androidTest/java/net/gini/android/bank/sdk/exampleapp/ui/testcases/MainActivityTest.kt

```diff
@@ -0,0 +1,117 @@
+package net.gini.android.bank.sdk.exampleapp.ui.testcases
+
+import android.Manifest
+import android.app.Activity
+import android.app.Instrumentation
+import android.content.Intent
+import android.net.Uri
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intending
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasAction
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasType
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.rule.GrantPermissionRule
+import net.gini.android.bank.sdk.exampleapp.ui.MainActivity
+import net.gini.android.bank.sdk.exampleapp.ui.screens.CaptureScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.HelpScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.MainScreen
+import net.gini.android.bank.sdk.exampleapp.ui.screens.OnboardingScreen
+import org.hamcrest.Matchers.allOf
+import org.junit.After
+import org.junit.Before
+import org.junit.FixMethodOrder
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class MainActivityTest {
+
+    @get: Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
+
+    @get: Rule
+    val grantPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.CAMERA)
+
+    private val mainScreen = MainScreen()
+    private val onboardingScreen = OnboardingScreen()
+    private val captureScreen = CaptureScreen()
+    private val helpScreen = HelpScreen()
+
+    @Before
+    fun setUp() {
+        Intents.init()
+    }
+    @Test
+    fun test1_assertTitlesOnMainScreen() {
+        mainScreen.checkWelcomeTitleIsDisplayed()
+        mainScreen.assertDescriptionTitle()
+    }
+    @Test
+    fun test2_clickPhotoPaymentButton() {
```

Comment:
[Request] It's not clear for me what this test does. When I read the test name I think it checks that the photo payment button can be clicked. But when I read the contents I see that it does much more than that and does assertions on three screens.

- As a rule of thumb a test should test only one thing and state that explicitly in its name.
- A popular practice is to have a _setup_, _action_ and _assert_ section (in this order) for each test. By avoiding mixing actions and assertions it is easier to read and maintain the tests.
- It's ok to have long test method names. The clearer it is what the test does the easier it is to find what could be wrong when tests fail.
- Tests should be independent so that we can run one specific test without having to run x nr of other tests. Use [clearPackageData](https://developer.android.com/training/testing/instrumented-tests/androidx-test-libraries/runner#enable-gradle) to clear the app state before each test (the code example is outdated there, use this instead `testInstrumentationRunnerArguments["clearPackageData"] = "true"`).

I would refactor this test into two tests and make them independent of eachother:
```
@Test
fun showsOnboardingOnFirstLaunchBeforeShowingTheCameraScreen() {
    // Action
    mainScreen.clickPhotoPaymentButton()
    // Assert
    onboardingScreen.assertSkipButtonText()
}

@Test
fun skipsOnboardinAndShowsCameraScreenOnSubsequentLaunches() {
    // Setup - launch the SDK and exit it
    mainScreen.clickPhotoPaymentButton()
    onboardingScreen.clickSkipButton()
    captureScreen.clickCancelButton()

    // Action - launch the SDK again
    mainScreen.clickPhotoPaymentButton()

    // Assert - SDK goes directly to the camera screen
    captureScreen.assertCameraTitle()
}
```

Reply:
I did separated the test-case into two, but for `test skipsOnboardinAndShowsCameraScreenOnSubsequentLaunches`, i am unable to Launch the SDK again and due to which the lines after `captureScreen.clickCancelButton()` is failing.
Can you help me with this @a-szotyori ?

Reply:
The method `test2_clickPhotoPaymentButton` is removed from `MainScreenTests` and is now a part of test class, `OnboardingScreenTests`